### PR TITLE
fix openwrt mips32 static build

### PR DESCRIPTION
**Pull request by XutaxKamay**
We did talk some months ago about it on IRC, just asking to pull request this for now, I'll figure out the rest for the blackhole thing maybe its due to kernel config idk, I'll let you know

### CODE REVIEW

**Comment by XutaxKamay** [comment]
Ah wait hold on, need to fix something

File: cmake/StaticBuild.cmake [hunk 1]
```diff
@@ -258,6 +259,10 @@ if(CMAKE_CROSSCOMPILING)
     set(openssl_arch linux-mips64)
   elseif(ARCH_TRIPLET STREQUAL mips-linux-gnu)
     set(openssl_arch linux-mips32)
+  elseif(ARCH_TRIPLET STREQUAL mips-openwrt-linux)
+    # needed to fix unbound so that it links correctly with openssl (crypto, EVP_sha256)
```
Comment (majestrate) [inline cmake/StaticBuild.cmake]: this comment seems like it'd be best in a commit message :^D

Comment (XutaxKamay) [reply cmake/StaticBuild.cmake]: okay xD

Comment (XutaxKamay) [reply cmake/StaticBuild.cmake]: just I leave the comment just in case though?

Comment (majestrate) [reply cmake/StaticBuild.cmake]: i think having it in the commit only is better so i git blame preserves the info

Comment (XutaxKamay) [reply cmake/StaticBuild.cmake]: oki

File: cmake/StaticBuild.cmake [hunk 2]
```diff
@@ -319,7 +323,7 @@ build_external(unbound
   --enable-static --with-libunbound-only --with-pic
   --$<IF:$<BOOL:${WITH_LTO}>,enable,disable>-flto --with-ssl=${DEPS_DESTDIR}
   --with-libexpat=${DEPS_DESTDIR}
-  "CC=${deps_cc}" "CFLAGS=${deps_CFLAGS}"
+  "CC=${deps_cc}" "CFLAGS=${deps_CFLAGS}" "LDFLAGS=${deps_latomic}"
```
Comment (majestrate) [inline cmake/StaticBuild.cmake]: better name: `deps_LDFLAGS`

**Comment by majestrate** [comment]
a better commit message would be

```
fix openwrt mips32 static build

libunbound needs to link against libatomic because of openssl
```

File: cmake/StaticBuild.cmake (hunk 2 above)
Comment (XutaxKamay) [reply cmake/StaticBuild.cmake]: actually it was named like this before, but i remember seeing deps_ld somewhere

Comment (XutaxKamay) [reply cmake/StaticBuild.cmake]: one sec

Comment (XutaxKamay) [reply cmake/StaticBuild.cmake]: okay I think it should be fine to use deps_ld cause it's only used for android stuffs which is before where it sets the variable, i'll just re-use it https://github.com/oxen-io/lokinet/blob/bf0dc52df7c05d4a14b5d5140615c1d547430055/cmake/StaticBuild.cmake#L254

**Comment by XutaxKamay** [comment]
> a better commit message would be
> 
> ```
> fix openwrt mips32 static build
> 
> libunbound needs to link against libatomic because of openssl
> ```

okay, let me know if it's good now

**Review by majestrate — APPROVED** [review]
looks good waiting for ci

**Comment by majestrate** [comment]
ci failed.

File: cmake/StaticBuild.cmake (hunk 2 above)
Comment (majestrate) [reply cmake/StaticBuild.cmake]: no, that's the path to the linker itself.

Comment (majestrate) [reply cmake/StaticBuild.cmake]: (we want to set additional linker flags)

**Comment by XutaxKamay** [comment]
ah, got link? can't see it here i think

**Comment by majestrate** [comment]
> ah, got link? can't see it here i think

ci.oxen.rocks

**Comment by majestrate** [comment]
https://ci.oxen.rocks/oxen-io/lokinet

**Comment by XutaxKamay** [comment]
> https://ci.oxen.rocks/oxen-io/lokinet

 LDFLAGS=/usr/lib/android-ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi-ld

okay yeah that's what you said earlier, let me fix it

**Review by majestrate — COMMENTED** [review]
okay this looks right, waiting for ci

**Comment by majestrate** [comment]
oh ci done. let's merge.

**Review by majestrate — APPROVED** [review]

**Comment by majestrate** [comment]
(ci status delivery hooks are borked yay)

**Comment by XutaxKamay** [comment]
ah great sorry went to eat

**Pull request merged**